test/events/component-view.test.js

```diff
@@ -0,0 +1,140 @@
+
+const config = {
+	context: {
+		product: 'desktop'
+	},
+	user: {
+		user_id: '123456'
+	}
+};
+
+function viewed (target) {
+	const event = new Event('intersect');
+	target.dispatchEvent(event);
+}
+
+function setMockIntersectionObserver (observeSpy, unobserveSpy) {
+	function mockIntersectionObserver(callback, options) {
+		this.callback = callback;
+		this.options = options;
+	}
+
+	mockIntersectionObserver.prototype.observe = function (target) {
+		target.addEventListener('intersect', () => {
+			this.callback([{ isIntersecting: true, target }]);
+		});
+		observeSpy(target);
+	};
+
+	mockIntersectionObserver.prototype.unobserve = function (target) {
+		unobserveSpy(target);
+	};
+
+	window.IntersectionObserver = mockIntersectionObserver;
+}
+
+
+describe('component:view', () => {
+
+	let targetComponent;
+	const observeSpy = sinon.spy();
+	const unobserveSpy = sinon.spy();
+
+	beforeEach(() => {
+		session.init();
+		send.init();
+		settings.set("config", config);
+		setMockIntersectionObserver(observeSpy, unobserveSpy);
+		sinon.spy(core, 'track');
+	});
+
+	afterEach(() => {
+		observeSpy.resetHistory();
+		unobserveSpy.resetHistory();
+		document.body.removeChild(targetComponent);
+		targetComponent = undefined;
+		core.track.restore();
+	});
+
+	context('with default props', () => {
+		beforeEach(() => {
+			targetComponent = document.createElement('div');
+			targetComponent.setAttribute('data-o-tracking-view', true);
+			targetComponent.innerHTML = 'Default component view elements';
+			document.body.appendChild(targetComponent);
+
+			componentView.init();
+			viewed(targetComponent);
+		});
+
+		it('should track an event for a component view ', () => {
+			assert.equal(observeSpy.calledOnce, true, "IntersectionObserver observed target");
+			assert.equal(unobserveSpy.calledOnce, true, "IntersectionObserver unobserved target");
+			assert.equal(core.track.calledOnce, true, "view event tracked");
+		});
+
+		it('should track with correct detail', () => {
+			assert.equal(core.track.getCall(0).args[0].category, 'component');
+			assert.equal(core.track.getCall(0).args[0].action, 'view');
+			assert.ok(core.track.getCall(0).args[0].context.domPathTokens, true);
+		});
+	});
+
+	context('with custom props', () => {
+		const category = 'custom';
+		const id = '1234';
+		const type = 'audio';
+		const optsCallbackSpy = sinon.spy();
+		const targetAttribute = 'data-custom-element';
+		const idAttribute = 'data-id';
+
+		beforeEach(() => {
+			targetComponent = document.createElement('div');
+			targetComponent.innerHTML = 'Custom component view elements';
+			targetComponent.setAttribute(targetAttribute, true);
+			targetComponent.setAttribute(idAttribute, id);
+			document.body.appendChild(targetComponent);
+
+			const opts = {
+				category,
+				elementsToTrack: [...document.querySelectorAll(`[${targetAttribute}]`)],
+				callback: (el) => {
+					optsCallbackSpy(el);
```

Comment:
Minor thing, but given you're asserting on the data returned by this function, you could probably eschew spying on this function as well.